ec2/subnets.go

```diff
@@ -0,0 +1,54 @@
+package ec2
+
+import (
+	"context"
+	"strings"
+
+	"github.com/YaleSpinup/apierror"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+func (e *Ec2) ListSubnets(ctx context.Context, vpc string) ([]map[string]string, error) {
+	if vpc != "" && !strings.HasPrefix(vpc, "vpc-") {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
```

Comment:
Maybe we can put "invalid input - wrong vpc prefix?" just something a touch more descriptive would help I think.